### DISCUSSION TITLE
umi 如何获取当前路由配置

### DISCUSSION THREAD

**Discussion by Yovven** [umijs/umi#7818] [Q&A]
按照文档中的介绍 [路由组件参数](https://umijs.org/zh-CN/docs/routing#%E8%B7%AF%E7%94%B1%E7%BB%84%E4%BB%B6%E5%8F%82%E6%95%B0)，在组件通过props.route可以获取到当前路由配置信息，但是打印出来还是整个路由配置，想拿到当前路由配置还需要再进一步筛选处理。求问有办法可以直接拿到当前路由配置吗？

**Comment by qiutian00** [comment]
`history.location`?

**Reply by Yovven** [reply]
history.location拿不到我自己再route中加的一些配置噢

**Reply by qiutian00** [reply]
噢噢， umi3可以 `import { history } from 'umi'`; 
![image](https://user-images.githubusercontent.com/24382500/151101136-c91caf76-9f5c-43c5-be6c-7675da468b21.png)

**Reply by Yovven** [reply]
是的，但是没有我自己在路由表中加的自定义配置

**Reply by YvieWong** [reply]
我也想知道,解决了吗?

**Reply by Yovven** [reply]
没有解决，只能遍历

**Reply by YvieWong** [reply]
> 

好吧,我也只能先这么干了

**Comment by Yovven** [comment]
邮件收到啦~~谢谢你！

**Reply by KCVO1995** [reply]
请问处了遍历，还有其他办法吗？

**Reply by cuijiudai** [reply]
咋搞啊？

**Reply by Yovven** [reply]
目前只能遍历，没有其他办法

**Reply by seafronthu** [reply]
不能立刻拿到当前路由和它的下级路由，得使用useAppData每次从头到尾遍历，性能也太差了
